src/algebra/ring/basic.lean

```diff
@@ -927,6 +927,37 @@ lemma neg_mul_eq_mul_neg (a b : α) : -(a * b) = a * -b :=
 lemma neg_mul_comm (a b : α) : -a * b = a * -b :=
 by simp
 
+/-- A type endowed with `-` and `*` has distributive negation, if it admits an injective map that
+preserves `-` and `*` to a type which has distributive negation. -/
+protected def function.injective.has_distrib_neg [has_neg β] [has_mul β] (f : β → α)
+  (hf : injective f) (neg :  ∀ a, f (-a) = -f a) (mul : ∀ a b, f (a * b) = f a * f b) :
```

Comment:
Is it any easier to say `f` is a `mul_hom_class`?

Reply:
No, because then you have to bundle `f` into a `mul_hom` on the fly and making `f` match `⇑⟨f, h⟩` breaks elaboration on dot notation.

Even when a single bundled hom holds several properties, I'm of the opinion that we should use unbundled functions because heuristically we either don't have the hom at the time of defining the instance or don't care about it enough to make it a standalone definition. @eric-wieser would probably disagree though.

Reply:
```suggestion
  (hf : injective f) (neg : ∀ a, f (-a) = -f a) (mul : ∀ a b, f (a * b) = f a * f b) :
```
I'm fine with leaving this as is, the pattern of passing in the hom is only present in the has_scalar hierarchy where typically you've already built all the single-variable typeclass instances.